genai_stack/model/gpt3_5.py

```diff
@@ -1,84 +1,152 @@
+
+
+class OpenAIGpt35ModelConfig(BaseModelConfig):
+    data_model = OpenAIGpt35ModelConfigModel
 
 
 class OpenAIGpt35Model(BaseModel):
-    model_name = "Gpt_3.5"
-    required_fields = ["openai_api_key"]
-
-    def load(self, model_path: str):
-        self.model = ChatOpenAI(
-            model_name=self.model_config_fields.get("model_name", "gpt-3.5-turbo-16k"),
-            openai_api_key=self.model_config_fields.get("openai_api_key"),
-            temperature=0,
-        )
-
-    def parse_chat_history(self, *args, **kwargs):
-        return "".join(" \n " + argument for argument in args)
-
-    def predict(self, query: str):
-        query = self.preprocess(query)
-        llm = self.model
-
-        if self.model_config.get("chat", False):
-            return self._vector_retreiver_qa(llm, query)
-        elif self.retriever:
-            return self._vector_retreiver_qa(llm, query)
-        else:
-            return self._without_retreiver_qa(llm, query)
-
-    def _without_retreiver_qa(self, llm, query):
-        template = """Question: {question}
-
-            Answer: """
-
-        prompt = PromptTemplate(template=template, input_variables=["question"])
-        llm_chain = LLMChain(prompt=prompt, llm=llm)
-        question = query
-
-        return llm_chain.run(question)
-
-    def _vector_retreiver_qa(self, llm, query):
-        conversation_chain = ConversationalRetrievalChain.from_llm(
-            llm=llm,
-            retriever=self.retriever.get_langchain_retriever(),
-            memory=self.get_memory(),
-            get_chat_history=self.parse_chat_history,
-            return_source_documents=True,
-        )
-        results = conversation_chain({"question": query})
-        return self.parse_chat_result(results)
-
-    def parse_chat_result(self, chat_result: dict):
-        return {
-            "result": chat_result["answer"],
-            "source_documents": self._parse_source_documents(
-                chat_result["source_documents"],
-            ),
-        }
-
-    def parse_qa_result(self, qa_result: dict):
-        return {
-            "result": qa_result["result"],
-            "source_documents": self._parse_source_documents(
-                qa_result["source_documents"],
-            ),
-        }
-
-    def parse_generations(self, generation_lst: list):
-        """
-        Recursively parse the text in the generations
-        """
-        result = """"""
+    config_class = OpenAIGpt35ModelConfig
 
-        for g in generation_lst:
-            if isinstance(g, list):
-                result += self.parse_generations(g)
-            if isinstance(g, Generation):
-                result += g.text
+    def _post_init(self, *args, **kwargs):
+        self.model = self.load()
 
-        return result
+    def load(self):
+        """
+        Using __dict__ here to dynamically access object attributes
+        """
+        model = ChatOpenAI(**self.config.parameters.__dict__)
```

Comment:
I guess there is a pydantic function called to_dict() or something can you check once instead of using the __dict__ directly

Reply:
No, it's not there to_dict doesn't work here I tried it. Actually to_dict is not predefined python function itself.
https://chat.openai.com/share/fcebe798-3d71-4149-8bf1-65e28d8254ac

Reply:
Refer this doc: https://docs.pydantic.dev/1.10/usage/exporting_models/